ui/app/AppLayouts/Chat/views/communities/CommunityNewPermissionView.qml

```diff
@@ -19,18 +19,137 @@ StatusScrollView {
 
     property var store
     property int viewWidth: 560 // by design
+    property bool isEditState: false
+    property bool dirty: {
+
+        let trick = d.triggerDirtyTool // Trick: Used to force the reevaluation of dirty when an item of the list is updated
+
+        // Holdings:
+        let dirtyHoldingsList = d.checkIfHoldingsDirty()
+
+        // Permissions:
+        let dirtyPermissionObj = false
+        if(root.permissionObject && d.dirtyValues.permissionObject.key !== null) {
+            dirtyPermissionObj = (d.dirtyValues.permissionObject.key !== root.permissionObject.key) ||
+                    (d.dirtyValues.permissionObject.text !== root.permissionObject.text) ||
+                    (d.dirtyValues.permissionObject.imageSource !== root.permissionObject.imageSource)
+        }
+        else {
+            dirtyPermissionObj = d.dirtyValues.permissionObject.key !== null
+        }
+
+        // TODO: Channels:
+        let dirtyChannelsList = false
+
+        return dirtyHoldingsList || dirtyPermissionObj || dirtyChannelsList || d.dirtyValues.isPrivateDirty
+    }
```

Comment:
The js code bloat could be significantly reduced by introducing C++ model wrapping original one. For better explanation there is an example of potential use:
```
IntermediateModel /*Or thing a better appropiate naming*/ {
   id: intermediateModel
   sourceModel: hodlingsModel
}
```
Such model would have the same roles of the provided sourceModel + additional dirty role. Moreover it could expose additional dirty property for the whole model set to true when there is any change. It could be used as follows:
```
Button {
  text: "Close"
  onClicked: {
     if (intermediateModel.dirty)
        // open confirm exit modal
     else
        closeView()
   }
}
```

Reply:
Of course it's better to handle it as a separate task: https://github.com/status-im/status-desktop/issues/8830

Reply:
Yeah, as discussed, it is an initial steps and will be improved in a separate task.